src/scripts/interactions.js

```diff
@@ -0,0 +1,216 @@
+import {
+  Raycaster,
+  Vector3, Matrix4, Quaternion
+} from 'three';
+import { getCurrentScene } from './currentScene';
+import { XR } from './xrController';
+import { camera } from './renderer/camera';
+
+// TODO: Split Interactions into indevidual interfaces:
+// - HoverInteraction
+// - DragInteraction
+// - SelectInteraction
+// Make different symbols for each so that instead of checking if an object has
+// interactions and then checking if it has the specific function, then we could
+// just check for the one symbol.
+
+// Put Interaction Callbacks under a special symbol
+// so that we don't get confused with things on the three Object
+export const Interactions = Symbol('Symbol Interactions');
+
+let inputSources = [];
+const handleInputSourcesChange = ({ session }) => {
+  inputSources = session.getInputSources();
+  console.log('Input Sources Changed.');
+};
+
+function createRay(inputSource, xrFrame) {
+  if (inputSource.targetRaySpace) {
+    const rayPose = xrFrame.getPose(inputSource.targetRaySpace, XR.refSpace);
+
+    if (rayPose) {
+      /* global XRRay:true */
+      return new XRRay(rayPose.transform);
+    }
+  }
```

Comment:
We need to be rendering in a laser line here if the input source has a controller associated with it. See my implementation.

Reply:
I can't troubleshoot the laser fully but I think it's working.